src/components/Property/RaiseWorkOrder/RaiseWorkOrderFollowOn/FollowOnLookup.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react'
+import ErrorMessage from '../../../Errors/ErrorMessage'
+import { DataList } from '../../../Form'
+import SpinnerWithLabel from '../../../SpinnerWithLabel'
+import { frontEndApiRequest } from '@/root/src/utils/frontEndApiClient/requests'
+import { WorkOrder } from '@/root/src/models/workOrder'
+
+interface Props {
+  visible: boolean
+  propertyReference: string
+  register: any
+  errors: { [key: string]: { message: string } }
+}
+
+const FollowOnLookup = (props: Props) => {
+  const { visible, propertyReference, register, errors } = props
+
+  const [workOrders, setWorkOrders] = useState<WorkOrder[]>([])
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const fetchRelatedWorkOrders = async () => {
+    if (workOrders.length > 0) {
+      // we dont need to fetch new data
+      return
+    }
```

Comment:
Is it possible that we already have some workorders, but not all?

Reply:
Not entirely sure what you mean. This only fetches the 5 most recent work orders as an auto suggestion. The user will be pasting in a work order reference anyway

Reply:
I was thinking whether it's possible to have that we have 3WOs loaded, when 5 are available or smth.
If yes, this condition wouldn't load the remaining 2.

But from what you're saying that's not possible, so all good.

Reply:
The number of related work orders arent going to change that frequently. Its not a race condition, if thats what youre suggesting